numpy/core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -2363,6 +2363,35 @@ run_intp_converter(PyObject* NPY_UNUSED(self), PyObject *args)
     return tup;
 }
 
+PyObject *
+test_scalar_intp_converter(PyObject *dummy, PyObject *obj)
+{
+    PyArray_Dims dummy_dims;
+    int output = PyArray_IntpConverter(obj, &dummy_dims);
```

Comment:
For converters here, you could also write:
```
if (!PyArray_IntpConverter(obj, &dummy_dims)) {
    return NULL;
}
/* continue here (no else if there is an error return) */
```

Reply:
I probably did not understand your review, I refactored the code to this form:
```c
PyObject *
test_scalar_intp_converter(PyObject *dummy, PyObject *obj)
{
    PyArray_Dims dummy_dims;
    if (!PyArray_IntpConverter(obj, &dummy_dims)) {
        return NULL;
    }
    else {
        return PyLong_FromLong(dummy_dims.ptr[0]);
    }
}
```

What do you mean by _no else if there is an error return_?

Reply:
This is fine, but you do not need the `else` clause at all, since the `if` already returns.  Preferences vary, although for short error-returns I think it is firmly on the side of removing the `else`.